e2e/Microsoft.Azure.Devices.E2ETests/DeviceTokenRefreshE2ETests.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Devices.Client;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    [TestClass]
+    public class DeviceTokenRefreshE2ETests
+    {
+        private const string DevicePrefix = "E2E_Message_TokenRefresh_";
+
+        [TestMethod]
+        public async Task DeviceClient_TokenIsRefreshed_Ok_Http()
+        {
+            await DeviceClient_TokenIsRefreshed_Ok_Internal(Client.TransportType.Http1);
+        }
+
+        [TestMethod]
+        public async Task DeviceClient_TokenIsRefreshed_Ok_Amqp()
+        {
+            await DeviceClient_TokenIsRefreshed_Ok_Internal(Client.TransportType.Amqp);
```

Comment:
need to test over websocket?

Reply:
Websocket doesn't change the way tokens are refreshed: it's only adding another network layer.